backend/housing.go

```diff
@@ -0,0 +1,72 @@
+// Copyright Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package backend
+
+import (
+	"net/http"
+	"math"
+	"strconv"
+	"fmt"
+)
+
+const (
+	ERROR_CASE_HOUSING = "error"
+	HOUSING_SAMPLE_PATH    = "/samples_templates/housing/"
+)
+
+func InitHousingForm() {
+	http.HandleFunc(HOUSING_SAMPLE_PATH+"calculate-mortgage-xhr", calculateMortgageXHR)
+	http.HandleFunc(HOUSING_SAMPLE_PATH+"calculate-mortgage", calculateMortgage)
+}
+
+func calculateMonthlyPayment(r *http.Request) float64 {
+	price, err := strconv.Atoi(r.FormValue("price"))
+	deposit, err := strconv.Atoi(r.FormValue("deposit"))
+	interest, err := strconv.ParseFloat(r.FormValue("annual_interest"), 64)
+	period,err := strconv.Atoi(r.FormValue("repayment_period"))
+
+	if err != nil {
```

Comment:
do you mean just returning it from the function or also add the error on the page so that the user sees it?